static/app/components/replays/player/__stories__/replaySlugChooser.tsx

```diff
@@ -14,33 +16,50 @@ export default function ReplaySlugChooser(props: Props) {
 
   const [replaySlug, setReplaySlug] = useSessionStorage('stories:replaySlug', '');
 
-  let content = null;
-  if (replaySlug) {
-    if (children) {
-      content = (
-        <ReplayLoadingState replaySlug={replaySlug}>
+  const input = (
+    <input
+      defaultValue={replaySlug}
+      onChange={event => {
+        setReplaySlug(event.target.value);
+      }}
+      placeholder="Paste a replaySlug"
+      css={css`
+        font-variant-numeric: tabular-nums;
+      `}
+      size={34}
+    />
+  );
+
+  if (replaySlug && children) {
+    function Content() {
+      const organization = useOrganization();
+      const readerResult = useLoadReplayReader({
+        orgSlug: organization.slug,
+        replaySlug,
+        clipWindow: undefined,
+      });
+      return (
+        <ReplayLoadingState readerResult={readerResult}>
           {({replay}) => <Providers replay={replay}>{children}</Providers>}
         </ReplayLoadingState>
       );
-    } else if (render) {
-      content = render(replaySlug);
     }
+    return (
+      <Fragment>
+        {input}
+        <Content />
+      </Fragment>
+    );
   }
 
-  return (
-    <Fragment>
-      <input
-        defaultValue={replaySlug}
-        onChange={event => {
-          setReplaySlug(event.target.value);
-        }}
-        placeholder="Paste a replaySlug"
-        css={css`
-          font-variant-numeric: tabular-nums;
-        `}
-        size={34}
-      />
-      {content}
-    </Fragment>
-  );
+  if (replaySlug && render) {
+    return (
+      <Fragment>
+        {input}
+        {render(replaySlug)}
```

Comment:
imo it's kind of confusing to have two different rendering methods in this component -- I don't really know how they're used but what if they were two components ReplaySlugChooser and ReplaySlugChooserWithLoaderStuff